### PR TITLE
16845: Adds parameters to return conditioned marginal stats, MINOR

### PR DESCRIPTION
**Pull request by cademack**
Add 'condition', 'precision', and 'num_cases' parameters to get_marginal_stats which allows users to get marginal statistics on a subset of the data that matches a given condition.

### CODE REVIEW

**Review by howsohazard — DISMISSED** [review]

File: trainee_template/marginal.amlg [hunk 1]
```diff
@@ -607,39 +646,39 @@
...
 											(retrieve_from_entity
 												(first (contained_entities
-													(list (query_max feature 1))
+													(append filtering_queries (query_max feature 1))
 												))
 												feature
 											)
 										mean
-											(compute_on_contained_entities (list
+											(compute_on_contained_entities (append filtering_queries
```
Comment (howsoRes) [inline trainee_template/marginal.amlg]: filtering queries should be on the next line, lined up with other queries

File: trainee_template/marginal.amlg [hunk 2]
```diff
@@ -607,39 +646,39 @@
...
+											(compute_on_contained_entities (append filtering_queries (query_quantile feature 0.25 weight_feature)))
 										percentile_75
-											(compute_on_contained_entities (list (query_quantile feature 0.75 weight_feature)))
+											(compute_on_contained_entities (append filtering_queries (query_quantile feature 0.75 weight_feature)))
 									))
 
 									(assign (assoc
 										mean_absdev
-											(compute_on_contained_entities (list
+											(compute_on_contained_entities (append filtering_queries
```
Comment (howsoRes) [inline trainee_template/marginal.amlg]: filtering queries should be on the next line, lined up with other queries - same with all the statements below as well. Generally, either they are all queries on one line, or each on a separate line.

File: trainee_template/marginal.amlg [hunk 3]
```diff
@@ -542,23 +542,59 @@
...
+					)
+
+					;get the stats for the specified weight_feature
+					(get featureMarginalStatsMap weight_feature)
+				)
 
-			;get the stats for the specified weight_feature
-			(get featureMarginalStatsMap weight_feature)
+				;else we have a condition
+				(declare
```
Comment (howsoRes) [inline trainee_template/marginal.amlg]: let

Comment (cademack) [reply trainee_template/marginal.amlg]: ahhh smart

**Review by howsoRes — APPROVED** [review]

**Pull request merged**